src/arch/util/machine-common-core.C

```diff
@@ -1631,6 +1631,17 @@ static INLINE_KEYWORD void AdvanceCommunication(int whenidle) {
     CommunicationServerXpmem();
 #endif
 
+#if CMK_USE_SHMEM
+    CmiIpcBlock* block;
+    auto* manager =
```

Comment:
Can this if-statement be lifted out of this function, by doing it once during startup (rather than checking every scheduler call)?

Reply:
This check is necessary only when Converse programs are compiled against an SMP build of Charm++ with CMI-SHMEM enabled. In such cases, this check will fail -- since Converse does not init the `coreIpcManager_` by default. This check could be eliminated if we change the `Cpv` variable to a `Csv` variable, in which case no initialization is required.